src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -73,3 +110,69 @@ def sbt_jar(name, **kwargs):
                           main='no.such.main.Main',
                           custom_rules=shader_rules,
                           **kwargs)
+
+    cls.register_jvm_tool(register,
+                          Zinc.ZINC_EXTRACTOR_TOOL_NAME,
+                          classpath=[
+                            JarDependency(org='org.pantsbuild',
+                                          name='zinc-extractor_2.11',
+                                          rev=Zinc._ZINC_VERSION)
+                          ])
+
+  def __init__(self, *args, **kwargs):
+    super(Zinc, self).__init__(*args, **kwargs)
+    self.set_distribution(jdk=True)
+
+  @memoized_property
+  def rebase_map_args(self):
+    """We rebase known stable paths in zinc analysis to make it portable across machines."""
+    rebases = {
+        self.dist.real_home: '/dev/null/java_home/',
+        get_buildroot(): '/dev/null/buildroot/',
+        self.get_options().pants_workdir: '/dev/null/workdir/',
+      }
```

Comment:
Maybe, but I also want attempts to actually read/write to these paths to fail in noticeable way. Will include pants in the name.